string[] actions
---
bool is_valid
